pennylane/ops/op_math/controlled.py

```diff
@@ -190,6 +186,100 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _get_special_ops():
```

Comment:
```suggestion
@functools.lru_cache
def _get_special_ops():
```

Potentially we can just construct the data once.